core/chains/evm/txmgr/eth_resender.go

```diff
@@ -30,7 +32,7 @@ const defaultResenderPollInterval = 5 * time.Second
 type EthResender struct {
 	orm       ORM
 	ethClient evmclient.Client
-	ks        KeyStore
+	ks        KeyStore[common.Address, *big.Int, gethTypes.Transaction, int64]
```

Comment:
I had the same idea, I floated it by the team - If everyone is okay Ill probably change it to using this way - better I think

Reply:
I think using `EthKeystore` would be useful in the cases where it will specifically be EVM types

imo I would leave the parts where eventually the generic type will be moved up a level - so for example, the `EthResender` will eventually contain the top level types that are passed to the `Keystore`. feels like duplicate work to  remove and then put back in later 🤷🏼 but either way would work 😄